servo-media/lib.rs

```diff
@@ -28,20 +29,22 @@ pub trait BackendInit {
 pub trait Backend: Send + Sync {
     fn create_player(
         &self,
+        id: MediaContextId,
         stream_type: StreamType,
         sender: IpcSender<PlayerEvent>,
         renderer: Option<Arc<Mutex<dyn FrameRenderer>>>,
         gl_context: Box<dyn PlayerGLContext>,
-    ) -> Box<dyn Player>;
+    ) -> Arc<Mutex<dyn Player>>;
     fn create_audiostream(&self) -> MediaStreamId;
     fn create_videostream(&self) -> MediaStreamId;
     fn create_stream_output(&self) -> Box<dyn MediaOutput>;
     fn create_audioinput_stream(&self, set: MediaTrackConstraintSet) -> Option<MediaStreamId>;
     fn create_videoinput_stream(&self, set: MediaTrackConstraintSet) -> Option<MediaStreamId>;
-    fn create_audio_context(&self, options: AudioContextOptions) -> AudioContext;
+    fn create_audio_context(&self, id: MediaContextId, options: AudioContextOptions) -> Arc<Mutex<AudioContext>>;
     fn create_webrtc(&self, signaller: Box<dyn WebRtcSignaller>) -> WebRtcController;
     fn can_play_type(&self, media_type: &str) -> SupportsMediaType;
     fn set_capture_mocking(&self, _mock: bool) {}
+    fn mute(&self, _val: bool) {}
```

Comment:
nit: this should probably be named `set_mute` to make it consistent with the `Mutable::set_mute` function.

Some documentation would also be really appreciated (i.e. mention that this method mute/unmute all Player and AudioContext instances).

Reply:
is leaving `mute` fn name in `Muteable` trait ok?
`set_mute` will conflict with `inner_player_proxy!(set_mute, val, bool);` of `impl Player for GStreamerPlayer`

Reply:
Yes, it's not a big deal. And actually, I was wrong asking this. I don't know why I thought `Muteable` had a `set_mute` function, but it is actually named `mute`. So just leave both methods as `mute`.

In any case, if you find yourself again in this scenario where you have to call methods with the same name, you can use https://doc.rust-lang.org/stable/book/ch19-03-advanced-traits.html#fully-qualified-syntax-for-disambiguation-calling-methods-with-the-same-name